### PR TITLE
[GH-1398] Fix `test-start-aou-workload` from polling forever

### PR DESCRIPTION
**Pull request by ehigham**
RR: https://broadinstitute.atlassian.net/browse/GH-1398
Split `workloads/when-done` into two functions:
- one that polls until the workload is :finished
- another that polls all the workflows
Add an assertion to the automation-test

### CODE REVIEW

File: api/test/wfl/system/automation_test.clj [hunk 1]
```diff
@@ -1,8 +1,8 @@
 (ns wfl.system.automation-test
-  (:require [clojure.pprint        :as pprint]
-            [clojure.test          :refer [deftest]]
+  (:require [clojure.test          :refer [deftest is]]
             [wfl.tools.fixtures    :as fixtures]
             [wfl.tools.resources   :as resources]
+            [wfl.service.cromwell  :refer [final-statuses]]
```
Comment (rexwangcc) [inline api/test/wfl/system/automation_test.clj:5]: We may want to define a new set of `active-statuses` for Firecloud workflows as it turns out they are different from the Cromwell's: `[ Queued, Launching, Submitted, Running, Failed, Succeeded, Aborting, Aborted, Unknown ]`?

**Review by rexwangcc — APPROVED** [review]
LGTM! (something's wrong with TDR dataset again…)

File: api/test/wfl/system/automation_test.clj (hunk 1 above)
Comment (ehigham) [reply api/test/wfl/system/automation_test.clj:5]: Created https://broadinstitute.atlassian.net/browse/GH-1410

**Comment by ehigham** [comment]
> LGTM! (something's wrong with TDR dataset again…)

Yep. They've got big stability issues.

File: api/test/wfl/tools/workloads.clj [hunk 2]
```diff
@@ -238,27 +238,33 @@
...
+  (let [finished? (let [skipped?    #(-> % :uuid util/uuid-nil?)
+                        terminated? (set cromwell/final-statuses)]
+                    (fn [{:keys [status] :as workflow}]
+                      (or (skipped? workflow) (terminated? status))))]
+    (done!
+     (util/poll
+      #(when (every? finished? (endpoints/get-workflows workload))
+         (endpoints/get-workload-status uuid))
+      polling-interval-seconds
+      max-polling-attempts))))
```
Comment (tbl3rd) [inline api/test/wfl/tools/workloads.clj:267]: Maybe `when-all-workflows-finish`?

**Review by tbl3rd — APPROVED** [review]
I guess we already have `test-append-to-aou-workload`.

**Pull request merged**